coe_code/format_output.py

```diff
@@ -0,0 +1,205 @@
+
+    local_logger.info("Loading AMP data.")
+    amp_filename = glob.glob(os.path.join(args.xml_files_path, 'f_amp2_*.xml'))[0]
+    with open(amp_filename, 'r', encoding='utf-8') as file:
+        amp_data = xmltodict.parse(file.read())
+
+    # Use dictionary comprehension to convert list of dictionaries into a single dictionary
+    amp_dict = {d['APID']: d for d in amp_data['ACTUAL_MEDICINAL_PRODUCTS']['AMPS']['AMP']}
+
+    return amp_dict
+
+
+def get_vmp_dict() -> dict:
+    ''' Function loads the VMP .xml file, and converts to a single dictionary where each key
+    is the VPID and the value is the dictionary of other values. '''
+
+    local_logger.info("Loading VMP data.")
+    vmp_filename = glob.glob(os.path.join(args.xml_files_path, 'f_vmp2_*.xml'))[0]
+    with open(vmp_filename, 'r', encoding='utf-8') as file:
+        vmp_data = xmltodict.parse(file.read())
+
+    # Use list comprehension to convert list of dictionaries into a single dictionary
+    vmp_dict = {d['VPID']: d for d in vmp_data['VIRTUAL_MED_PRODUCTS']['VMPS']['VMP']}
+
+    return vmp_dict
+
+
+def get_vtm_dict() -> dict:
+    ''' Function loads the VTM .xml file, and converts to a single dictionary where each key
+    is the VTMID and the value is the dictionary of other values. '''
+
+    local_logger.info("Loading VTM data.")
+    vtm_filename = glob.glob(os.path.join(args.xml_files_path, 'f_vtm2_*.xml'))[0]
+    with open(vtm_filename, 'r', encoding='utf-8') as file:
+        vtm_data = xmltodict.parse(file.read())
+
+    # Use list comprehension to convert list of dictionaries into a single dictionary
+    vtm_dict = {d['VTMID']: d for d in vtm_data['VIRTUAL_THERAPEUTIC_MOIETIES']['VTM']}
+
+    return vtm_dict
+
+
+def load_and_pivot_input() -> pd.DataFrame:
+    ''' Function loads a .csv matched input file from the medicines-text-mining-tool and pivots
+    the dataframe '''
+
+    # Use pandas to load the .csv matched input file from the medicines-text-mining-tool
+    local_logger.info("Loading input dataframe - %s.", args.input_file)
+    with open(args.input_file, 'r', encoding='utf-8') as file:
+        results = pd.read_csv(file, dtype=str)
+
+    # Pivot results to make a new column for each unique entry in the id_level column
+    # (APID, VPID, VMTID) and add the match_id value in the correct one
+    pivoted_results = results.pivot(
+        index=['original_epma_description', 'match_term', 'match_level'],
+        columns='id_level',
+        values='match_id'
+        )
+
+    return pivoted_results
+
+
+def get_vmp_from_amp(apid, amp_dict):
+    ''' Function to look up the VPID corresponding to an APID '''
+
+    if apid in amp_dict:
+        return amp_dict[apid].get('VPID', np.nan)
+    else:
+        return np.nan
+
+
+def get_vtm_from_vmp(vpid, vmp_dict):
+    ''' Function to look up the VPID and return a VTMID '''
+
+    if vpid in vmp_dict:
+        return vmp_dict[vpid].get('VTMID', np.nan)
+    else:
+        return np.nan
```

Comment:
cool -- some of the linters will tell you that the else is unnecessary, but I don't mind (i.e. if you just had return np.nan without the else and wihout the indent, you'd get exactly the same behavjour.